### DISCUSSION TITLE
Animating the terminal

### DISCUSSION THREAD

**Discussion by sketch34** [gabime/spdlog#2529] [Q&A] [answered]
Does spdlog (or fmt) have any support for having animated / updated output in the terminal?

I mean something like `top` on linux, where the terminal output will update within a given region. I'd love to add some metrics that are always visible in a certain region of the terminal as the regular logging scrolls past.

Probably [seq](https://datalust.co/seq) integration is a better approach. But I'm curious anyway.

![135775594-dd729ae2-22d7-4692-92e9-c174c29991b2](https://user-images.githubusercontent.com/1082293/199524564-a7890714-1bf4-4dbe-9d8c-5abad8feb0b1.gif)

**Comment by tt4g** [comment, marked as answer]
Color output with [ANSI escape code](https://en.wikipedia.org/wiki/ANSI_escape_code) is supported, but not animation.

In many cases, it is necessary to continually output ANSI escape code to STDOUT (STDERR) for animation.
It is not easy to manage animation timing, so it is recommended to use a library that specializes in this area.
